File: molecular/src/test/java/com/hartwig/actin/molecular/orange/evidence/known/TestKnownEventResolverFactory.java
package com.hartwig.actin.molecular.orange.evidence.known;

import com.hartwig.serve.datamodel.ImmutableKnownEvents;
import com.hartwig.serve.datamodel.KnownEvents;

import org.jetbrains.annotations.NotNull;

public final class TestKnownEventResolverFactory {

    private TestKnownEventResolverFactory() {
    }

    @NotNull
    public static KnownEventResolver createEmpty() {
        return new KnownEventResolver(ImmutableKnownEvents.builder().build());
    }

    @NotNull
    public static KnownEventResolver createProper() {
        KnownEvents knownEvents = ImmutableKnownEvents.builder()
                .addHotspots(TestServeKnownFactory.hotspotBuilder().build())
                .addCodons(TestServeKnownFactory.codonBuilder().build())
                .addExons(TestServeKnownFactory.exonBuilder().build())
                .addGenes(TestServeKnownFactory.geneBuilder().build())
                .addCopyNumbers(TestServeKnownFactory.copyNumberBuilder().build())
                .addFusions(TestServeKnownFactory.fusionBuilder().build())
                .build();

        return new KnownEventResolver(knownEvents);
    }
}

File: molecular/src/test/kotlin/com/hartwig/actin/molecular/evidence/known/TestKnownEventResolverFactory.kt
package com.hartwig.actin.molecular.evidence.known

import com.hartwig.serve.datamodel.ImmutableKnownEvents
import com.hartwig.serve.datamodel.KnownEvents

object TestKnownEventResolverFactory {

    fun createEmpty(): KnownEventResolver {
        return KnownEventResolver(ImmutableKnownEvents.builder().build(), emptySet())
    }

    fun createProper(): KnownEventResolver {
        val knownEvents: KnownEvents = ImmutableKnownEvents.builder()
            .addHotspots(TestServeKnownFactory.hotspotBuilder().build())
            .addCodons(TestServeKnownFactory.codonBuilder().build())
            .addExons(TestServeKnownFactory.exonBuilder().build())
            .addGenes(TestServeKnownFactory.geneBuilder().build())
            .addCopyNumbers(TestServeKnownFactory.copyNumberBuilder().build())
            .addFusions(TestServeKnownFactory.fusionBuilder().build())
            .build()

        return KnownEventResolver(knownEvents, knownEvents.genes())
    }
}
